hazelcast-spring/src/test/java/com/hazelcast/spring/TestClientDefaults.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spring;
+
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.client.impl.ClientTestUtil;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.springframework.test.context.ContextConfiguration;
+
+import javax.annotation.Resource;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(CustomSpringJUnit4ClassRunner.class)
+@ContextConfiguration(locations = {"client-defaults-context.xml"})
+@Category(QuickTest.class)
+public class TestClientDefaults {
+
+    private ClientConfig clientConfig;
+
+    @Resource(name = "client")
+    private HazelcastInstance client;
+
+    @BeforeClass
+    @AfterClass
+    public static void start() {
+        Hazelcast.shutdownAll();
+    }
+
+    @Before
+    public void before() throws IllegalAccessException {
+        clientConfig = ClientTestUtil.getClientConfig(client);
```

Comment:
@jerrinot You should be able to inject `HazelcastClientProxy` in this class (see [here](https://github.com/hazelcast/hazelcast/blob/master/hazelcast-spring/src/test/java/com/hazelcast/spring/TestClientApplicationContext.java#L77)) and then get the `ClientConfig` directly from `HazelcastClientProxy` (see [here](https://github.com/hazelcast/hazelcast/blob/master/hazelcast-spring/src/test/java/com/hazelcast/spring/TestClientApplicationContext.java#L155)). 
So, I think there is no need to use reflection magic.